### PR TITLE
ODC-7296: Rename all instances of YAMLEditor to CodeEditor

### PR DESCRIPTION
**Pull request by Lucifergene**
**Fixes**: 
https://issues.redhat.com/browse/ODC-7296

**Analysis / Root cause**: 
Current YAMLEditor also supports other languages like JSON. Therefore need to rename the component.

**Solution Description**: 
Renamed all instances of YAMLEditor to CodeEditor

**Unit test coverage report**: 
No Changes

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/docs/api.md [hunk 1]
```diff
@@ -65,6 +65,7 @@
 63. [DEPRECATED] [`PerspectiveContext`](#perspectivecontext)
 64. [DEPRECATED] [`useAccessReviewAllowed`](#useaccessreviewallowed)
 65. [DEPRECATED] [`useSafetyFirst`](#usesafetyfirst)
+66. [DEPRECATED] [`YAMLEditor`](#yamleditor)
```
Comment (jerolimov) [inline frontend/packages/console-dynamic-plugin-sdk/docs/api.md:68]: :+1:

**Review by jerolimov — APPROVED** [review]
The update looks good to me. Export as YAMLEditor still exists now.

/lgtm
/approve

**Comment by jerolimov** [comment]
Apply labels based on the epic/feature:

/label px-approved
/label docs-approved

Sanket, I don't know how we could test the plugin SDK export, but you maybe can test this code editor;

1. YAML add
2. YAML edit
3. a Dockerfile editor is displayed when create a BuildConfig and select "Dockerfile" as Source>Source type
4. a JSON editor is only  in the new "test serverless function" modal that is only available when you change the local storage value for `KNATIVE_SERVERLESS_FUNCTION_TEST` to `"true"`.

/cc @sanketpathak

**Comment by sanketpathak** [comment]
Verified on a cluster using cluster-bot. Tested the following cases :

- YAML add
- YAML edit
- a Dockerfile editor is displayed when create a BuildConfig and select "Dockerfile" as Source>Source type
-  tested json with broker/channel yaml editor
Screenshot from 2023-05-04 02-47-49
/label qe-approved

**Pull request merged**